backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,259 @@
+                userManager,
+                context, cancellationToken).ConfigureAwait(false))
+        {
+            return new CreateApplicationPayload(
+                new CreateApplicationError(
+                    CreateApplicationErrorCode.UNAUTHORIZED,
+                    "You are not authorized to create applications.",
+                    new[] { nameof(input), nameof(input.ClientId).FirstCharToLower() }
+                )
+            );
+        }
+        var descriptor = new OpenIddictApplicationDescriptor
+        {
+            ClientId = input.ClientId,
+            ClientSecret = "application.ClientSecret",
+            DisplayName = input.DisplayName,
+            ConsentType = environment.IsEnvironment(Program.TestEnvironment)
+                        ? OpenIddictConstants.ConsentTypes.Systematic
+                        : OpenIddictConstants.ConsentTypes.Explicit,
+            RedirectUris =
+            {
+                new Uri(environment.IsEnvironment(Program.TestEnvironment)
+                    ? "urn:test"
+                    : input.RedirectUri,
+                    UriKind.Absolute)
+            },
+            PostLogoutRedirectUris =
+            {
+                new Uri(environment.IsEnvironment(Program.TestEnvironment)
+                    ? "urn:test"
+                    : input.PostLogoutRedirectUri,
+                    UriKind.Absolute)
+            },
+            Permissions =
+            {
+                // Add default permissions
+                OpenIddictConstants.Permissions.Endpoints.Authorization,
+                OpenIddictConstants.Permissions.Endpoints.Introspection,
+                OpenIddictConstants.Permissions.Endpoints.Logout,
+                OpenIddictConstants.Permissions.Endpoints.Revocation,
+                OpenIddictConstants.Permissions.Endpoints.Token,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.GrantTypes.Password
+                    : OpenIddictConstants.Permissions.GrantTypes.AuthorizationCode,
+                OpenIddictConstants.Permissions.GrantTypes.RefreshToken,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.ResponseTypes.Token
+                    : OpenIddictConstants.Permissions.ResponseTypes.Code,
+                OpenIddictConstants.Permissions.Scopes.Address,
+                OpenIddictConstants.Permissions.Scopes.Email,
+                OpenIddictConstants.Permissions.Scopes.Phone,
+                OpenIddictConstants.Permissions.Scopes.Profile,
+                OpenIddictConstants.Permissions.Scopes.Roles,
+            },
+            Requirements =
+            {
+                OpenIddictConstants.Requirements.Features.ProofKeyForCodeExchange
+            }
+        };
+
+        foreach (var permission in JsonSerializer.Deserialize<List<string>>(input.Permissions) ?? new List<string>())
```

Comment:
Why is `input.Permissions` a `string` with `JSON` instead of `List` of `string`s? GraphQL supports string arrays, so there should be no need to JSON-encode a list of strings as a `string`. Doing so is rather error-prone.